src/machinetalk/messagebus/messagebus.cc

```diff
@@ -266,33 +266,33 @@ static int zmq_setup(msgbusd_self_t *self)
 {
     GOOGLE_PROTOBUF_VERIFY_VERSION;
 
-    // suppress default handling of signals in zctx_new()
+    // suppress default handling of signals in zsock_new()
     // since we're using signalfd()
     zsys_handler_set(NULL);
 
-    self->context = zctx_new ();
+    self->context = zsock_new ();
     assert(self->context);
 
-    zctx_set_linger (self->context, 0);
+    zsock_set_linger (self->context, 0);
 
-    self->cmd = zsocket_new (self->context, ZMQ_XPUB);
+    self->cmd = zsock_new (self->context, ZMQ_XPUB);
```

Comment:
This call here is indeed weird, there shouldn't be a context passed.

Reply:
If you look at the commit in my repo
https://github.com/ArcEye/machinekit/commit/3e489ab39fb574cd09a07306f37b2b38b24c3ca5
I think things like that are fixed, but it still does not run.